src/routing/route.cc

```diff
@@ -190,31 +200,164 @@ LaneSRangeRelation Route::ComputeLaneSRangeRelation(const api::LaneSRange& lane_
       get_lane_s_range_road_position(lane_s_range_b_index->first, lane_s_range_b_index->second, kEnd);
 
   return kRelativePositionToPreceedingLaneSRange[static_cast<size_t>(
-      ComputeRelativePositionFor(lane_s_range_a_road_pos, lane_s_range_b_road_pos, tolerance))];
+      ComputeRelativePosition(lane_s_range_a_road_pos, lane_s_range_b_road_pos, tolerance))];
 }
 
-std::optional<Route::LaneSRangeIndex> Route::FindLaneSRangeIndex(const api::LaneSRange& lane_s_range) const {
-  auto is_lane_s_range_contained = [tolerance = road_network_->road_geometry()->linear_tolerance()](
-                                       const api::LaneSRange& lane_s_range_a, const api::LaneSRange& lane_s_range_b) {
-    if (lane_s_range_a.lane_id() != lane_s_range_b.lane_id()) {
-      return false;
-    }
-    const bool s1_is_in_range = lane_s_range_a.s_range().s1() + tolerance >= lane_s_range_b.s_range().s1() &&
-                                lane_s_range_a.s_range().s0() - tolerance <= lane_s_range_b.s_range().s1();
-    const bool s0_is_in_range = lane_s_range_a.s_range().s1() + tolerance >= lane_s_range_b.s_range().s0() &&
-                                lane_s_range_a.s_range().s0() - tolerance <= lane_s_range_b.s_range().s0();
-    return s1_is_in_range && s0_is_in_range;
+api::LaneSRoute Route::ComputeLaneSRoute(const api::RoadPosition& start_position) const {
+  MALIPUT_THROW_UNLESS(start_position.lane != nullptr);
+
+  const api::RoadPosition& end_road_position = end_route_position();
```

Comment:
Consider renaming "end_road_position" to be "end_position" for brevity and clarity. It is not the end of the road. Since it is in class `Route`, it should be clear that "end_position" refers to the end of the route.

Reply:
Done.